### ISSUE TITLE
FURB118, `lambda x: x[:, 0]` should be `itemgetter((slice(None), 0))`, instead of `itemgetter((:, 0))`

### ISSUE THREAD

**Issue by nasyxx** [astral-sh/ruff#13508]
In refurb, FURB118, `lambda x: x[:, 0]` should be `itemgetter((slice(None), 0))`, instead of `itemgetter((:, 0))`

```python
# xx.py
import numpy as np
xs = np.array([[1, 2, 3]])
xm = map(lambda x: x[:, 0], xs)
```

```
xx.py:
  3:10 FURB118 [*] Use `operator.itemgetter((:, 0))` instead of defining a lambda
```

I guess `(:, 0)` is not a valid tuple.

**Comment by MichaReiser** [comment]
Running the above (after changing `np.array` to a normal list) gives me

```python

**Comment by nasyxx** [comment]
> @nasyxx could you clarify. Do you expect the above to be valid code?

Hi, the above is valid, while it will raise IndexError.  

My actual ndarray is quite complex, you can change `xs` to `xs = np.ones((5, 4, 3, 2, 1))`.  Then when running `list(xm)` will give you result.

**Comment by AlexWaygood** [comment]
There's definitely a serious bug here: we currently suggest invalid syntax (and, if you run with `--fix`, we introduce invalid syntax).

```diff
- lambda x: x[:, 0]
+ import operator
+ operator.itemgetter((:, 0))
```

In my opinion, `itemgetter((slice(None), 0))` (which would be the correct way of rewriting this with `itemgetter`, as @nasyxx says) is quite ugly, so I think I'd vote for not emitting a diagnostic at all in this case. But others may feel differently; emitting a diagnostic would be _consistent_ with the overall (quite opinionated) philosophy of `FURB118`.

**Comment by zanieb** [comment]
We already handle converting `x[:]` to `slice(None)` so I opted to continue emitting the diagnostic.
